### PR TITLE
introduce the IndexFeature for user

### PR DESCRIPTION
**Pull request by LHT129**
issue: #154

### CODE REVIEW

File: include/vsag/index_feature.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    SUPPORT_BATCH_SEARCH,
+    SUPPORT_METRIC_TYPE_L2,
+    SUPPORT_METRIC_TYPE_INNER_PRODUCT,
+    SUPPORT_METRIC_TYPE_COSINE,
+    SUPPORT_SERIALIZE_FILE,
+    SUPPORT_SERIALIZE_BINARY_SET,
+    SUPPORT_DESERIALIZE_FILE,
+    SUPPORT_DESERIALIZE_BINARY_SET,
+    SUPPORT_DESERIALIZE_READER_SET,
+    SUPPORT_RESET,
```
Comment (wxyucs) [inline include/vsag/index_feature.h]: what is this feature RESET?

File: include/vsag/index_feature.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    SUPPORT_SEARCH_ONE_WITH_MULTI_THREAD,
+    SUPPORT_BATCH_SEARCH_WITH_MULTI_THREAD,
+
+    SUPPORT_ADD_CONCURRENT,
+    SUPPORT_SEARCH_CONCURRENT,
+    SUPPORT_DELETE_CONCURRENT,
+    SUPPORT_ADD_SEARCH_CONCURRENT,
+    SUPPORT_ADD_DELETE_CONCURRENT,
+    SUPPORT_SEARCH_DELETE_CONCURRENT,
+    SUPPORT_ADD_SEARCH_DELETE_CONCURRENT,
```
Comment (wxyucs) [inline include/vsag/index_feature.h]: should we make all of them true in the future?

File: include/vsag/index_feature.h (hunk 1 above)
Comment (LHT129) [reply include/vsag/index_feature.h]: clear all data in index

File: include/vsag/index_feature.h (hunk 2 above)
Comment (LHT129) [reply include/vsag/index_feature.h]: not for all, some index may static without add and search concurrently. some feature can be enable by pass different params to factory. but may bring some performance loss. so user can disable these feature, just avoiding the concurrent between such behaviors in their own process with vsag

File: include/vsag/index_feature.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+namespace vsag {
+
+enum class IndexFeature {
```
Comment (jiaweizone) [inline include/vsag/index_feature.h:20]: IndexFeature -> IndexFeatures ?

Comment (LHT129) [reply include/vsag/index_feature.h:20]: easily misunderstood as a collection, it is actually an enum class. We will create another method to check which features in current feature collection.

**Review by jiaweizone — APPROVED** [review]
LGTM

**Pull request merged**